library/src/scala/IArray.scala

```diff
@@ -535,4 +648,67 @@ object IArray {
     // The double type ascription is currently needed,
     // for some reason (see: https://scastie.scala-lang.org/sSsmOhKxSKym405MgNRKqQ)
     Array.unapplySeq((x: Sup[T]): Array[_ <: T])
-}
+
+  // TODO where should this class be defined? The WithFilter for Array is in ArrayOps.
+  /** A lazy filtered array. No filtering is applied until one of `foreach`, `map` or `flatMap` is called. */
+  class WithFilter[T](p: T => Boolean, xs: IArray[T]):
```

Comment:
Now I also have `IArrayBuilder`

Reply:
The `WithFilter` of `Array` is in `ArrayOps` because that's where the `def withFilter` method is. In this case, since I guess `withFilter` is in `IArray`, there's no reason to put `WithFilter` anywhere else.

Reply:
Then I guess we dot the same for `IArrayBuilder`